### ISSUE TITLE
gatsby 2.30.0 to 2.31.0 breaks .eslintrc.js

### ISSUE THREAD

**Issue by theskillwithin** [gatsbyjs/gatsby#29133]
## Description

https://github.com/gatsbyjs/gatsby/commit/5f0d2146b591a0c09576c455f2f4f7e1424f8a20

### Steps to reproduce

Upgrade to gatsby 2.31.1 from 2.31.0 and .eslintrc.js file is ignored

```
  System:
    OS: macOS 11.1
    CPU: (12) x64 Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.3 - ~/.nvm/versions/node/v14.15.3/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.11 - ~/.nvm/versions/node/v14.15.3/bin/npm
  Languages:
    Python: 2.7.17 - /usr/local/bin/python
  Browsers:
    Chrome: 88.0.4324.96
    Firefox: 84.0
    Safari: 14.0.2
  npmPackages:
    gatsby: ^2.31.1 => 2.31.1
    gatsby-image: ^2.10.0 => 2.10.0
    gatsby-plugin-canonical-urls: ^2.9.0 => 2.9.0
    gatsby-plugin-disqus: ^1.2.2 => 1.2.2
    gatsby-plugin-google-tagmanager: ^2.10.0 => 2.10.0
    gatsby-plugin-local-search: ^2.0.0 => 2.0.0
    gatsby-plugin-manifest: ^2.11.0 => 2.11.0
    gatsby-plugin-offline: ^3.9.0 => 3.9.0
    gatsby-plugin-postcss: ^3.6.0 => 3.6.0
    gatsby-plugin-react-helmet: ^3.9.0 => 3.9.0
    gatsby-plugin-robots-txt: ^1.5.5 => 1.5.5
    gatsby-plugin-s3: ^0.3.8 => 0.3.8
    gatsby-plugin-sass: ^3.1.0 => 3.1.0
    gatsby-plugin-sharp: ^2.13.1 => 2.13.1
    gatsby-plugin-sitemap: ^2.11.0 => 2.11.0
    gatsby-plugin-svgr: ^2.1.0 => 2.1.0
    gatsby-plugin-webpack-bundle-analyser-v2: ^1.1.18 => 1.1.18
    gatsby-source-contentful: ^3.1.3 => 3.1.3
    gatsby-source-filesystem: ^2.10.0 => 2.10.0
    gatsby-transformer-remark: ^2.15.0 => 2.15.0
    gatsby-transformer-sharp: ^2.11.0 => 2.11.0
```

**Comment by hendra-go** [comment]
Confirmed. I also experience the same issue.

**Comment by mikaelkristiansson** [comment]
Same here

**Comment by icy-meteor** [comment]
Was breaking my head over this. Same here :)

**Comment by jablonski** [comment]
Same issue here, prevents us from updating to 2.31.x

**Comment by vladar** [comment]
Fixed in `gatsby@2.32.1`
